### PR TITLE
feat: switch to model-sdk for data-model rendering

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Switch to `generator-model-sdk` for data-model rendering:
- at the moment changed is only `$$objectSchema$$.java` template - related to model created from given schema
- I created `sample` folder with models to show new model and old -> you can check it here https://github.com/asyncapi/java-spring-template/tree/25dd21245aafff03a8e0726d7dedf8fe65168fe5/sample/src/main/java/com/asyncapi/model First is rendered new model, then old. You can make comparison. Of course it will be removed before merge - also `sample.yaml` spec.
- Unfortunately PR is blocked by https://github.com/asyncapi/generator/pull/516 and https://github.com/asyncapi/generator-model-sdk/pull/112 First is blocker, the code in second can be use internally in this repo, but I want to split some generic (and useful) functionality to proper place - maybe I will include it in PR.

Some improvements:
- render `@Min`, `@Max` annotation if value is set to `0`. Currently isn't rendered
- render `exclusiveMaximum` as `exclusiveMaximum - 1`, not `exclusiveMaximum + 1` - I think this is a bug.

Thoughts: 
- I have included rendering for `equal`, `hash`, `toString` function inside filter - maybe it should be moved as preset to `model-sdk` library.

Ref: https://github.com/asyncapi/generator-model-sdk/blob/master/docs/customisation.md#preset

Feel free to comments! :)

**Related issue(s)**
Related to https://github.com/asyncapi/shape-up-process/issues/49

### CODE REVIEW

**Comment by Tenischev** [comment]
Hey @magicmatatjahu if you need any help from my side, just say

**Comment by magicmatatjahu** [comment]
@Tenischev Hi! Thanks for caring, but at the moment no. I will try to open PR to review tomorrow, and then I will need a review from you. I just change the `model/$$objectSchema$$.java` to render the model with our library. There is a lot missing in it, so it will only be a small demonstration of the possibilities :)

**Comment by magicmatatjahu** [comment]
@Tenischev PR is open to review :)

File: package.json
Comment (jonaslagoni) [inline package.json:35]: Must be changed 😄

File: filters/model.js
Comment (jonaslagoni) [inline filters/model.js:314]: I would love some comments here 😅

Comment (jonaslagoni) [reply filters/model.js:314]: Someone who dont know what the model gen library is will have a hard time understanding this code 😄

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Just some general feedback 😄

File: package.json
Comment (magicmatatjahu) [reply package.json:35]: I use some presets from this PR https://github.com/asyncapi/generator-model-sdk/pull/112 and I have symlink here to use them :) Probably I will move presets to this repo, but don't worry, I remember about it.

File: filters/model.js
Comment (magicmatatjahu) [reply filters/model.js:314]: Yes, I will write, thanks! :)

File: filters/model.js
Comment (fmvilas) [inline filters/model.js:90]: Is it like this maybe?

```suggestion
const INLINE_ANY_OF = {
```

File: filters/model.js
Comment (fmvilas) [reply filters/model.js:314]: I have the feeling that most of the code here (if not all) should be extracted to its own preset module somewhere. If we want to do the same in the other Java template we have in the future, this code will remain the same and we don't want to duplicate it.

Also, remember, the purpose of the model generation library is to make it easier for template authors, and this is not easy :)

**Review by fmvilas — CHANGES_REQUESTED** [review]

File: package.json
Comment (fmvilas) [reply package.json:35]: Get used to `npm link`. This way you don't have to change things here: https://docs.npmjs.com/cli/v6/commands/npm-link.

File: filters/model.js
Comment (magicmatatjahu) [reply filters/model.js:314]: @fmvilas Yes I know about common code/presets which we can use in another templates - I created PR with `description`, `jackson`, `constraints` presets. I can also create presets for `hashCode`, `isEqual`, `toString` functions as well. Main problem in this model-generation is that I wanna to have this same way of rendering model as old one, so I had to make this magic with `INLINE_ANY_OF` etc. On the other hand, the presets themselves showed that you can change what you want in the model - of course you must know how it works, which interfaces presets implements etc, but yes, you can do that.

Comment (fmvilas) [reply filters/model.js:314]: I'm curious if @Tenischev would be happy without these. tweaks. Maybe we don't need to reproduce exactly what was there and leave exactly what the model generator is outputting.

**Comment by magicmatatjahu** [comment]
@jonaslagoni I added docs for functions and remove unnecessary code -> presets and naming convention simplified the model twice :)

**Comment by jonaslagoni** [comment]
@magicmatatjahu It is not my template so I don't feel like I should be the one to do the final review 🙂 

cc @Tenischev

**Comment by derberg** [comment]
@magicmatatjahu did you decide to discontinue this one?

**Comment by magicmatatjahu** [comment]
@derberg Yes, I didn't get a review and I'm not going back to it, but I'm not angry due to this situation :)

**Comment by derberg** [comment]
You did get a review as I see there were comments. We do not have a core maintainer here yet other than us, and I see that @Tenischev was pinged. We should be able to go forward.

> but I'm not angry due to this situation

I did not even consider that you could be angry, you are such a nice guy that I can't imagine you know what anger is 😄

**Pull request closed without merging**
